### DISCUSSION TITLE
Intercepting DLL Methods, is it possible?

### DISCUSSION THREAD

**Discussion by 1matin** [dotnet/roslyn#76468] [Q&A] [answered]
I am working on a .NET 9 Console Application with a NuGet package. As far as I know, NuGet packages are essentially DLL files that are cached on my PC and added to my project as a reference.

The package includes a specific method that creates an HttpClient and provides it to other parts of the library. I want to set the `DefaultProxyCredentials` for that HttpClient. Currently, I have achieved this by obtaining the source code of the package and manually modifying it. I then reference the modified `.csproj` file instead of the original NuGet package.

While this solution works, I'm looking for a way to accomplish this using the unmodified package from NuGet. This way, I could benefit from future updates without having to manually fetch the repository each time.

I am aware of the preview interceptors feature introduced in C# 12, but I understand that it requires the code's specific line and column, in order to intercept that call. How can this work with a compiled DLL file?

Thank you for your help!

**Comment by 333fred** [comment]
> How can this work with a compiled DLL file?

It cannot. This is not possible with Roslyn.

**Comment by huoyaoyuan** [comment, marked as answer]
Interceptors apply to source code before they are emitted into assemblies. To modify the compilation results, IL post-processing approaches like [Fody](https://github.com/Fody) are required.
